src/types.ts

```diff
@@ -1672,12 +1673,15 @@ export type Device<UserType = UnknownType> = DeviceFields & {
   user_id?: string;
 };
 
-export type DeviceFields = {
+export type BaseDeviceFields = {
+  id: string;
+  push_provider: 'apn' | 'firebase';
+};
+
+export type DeviceFields = Partial<BaseDeviceFields> & {
```

Comment:
```suggestion
export type DeviceFields = BaseDeviceFields & {
```

@ferhatelmas should these two fields always be required even? looks like the only usage is via `getDevices()`

Reply:
suggested change is good